design/specs/table-import.md

```diff
@@ -14,153 +14,175 @@ Adding a new table from a file import allows users to input data into Mathesar w
-Using the lookup field, the user could also add a new schema from this dialog.
+### Uploading the Data File
 
-![](/assets/design/specs/table-import/ByfztC0tu.png =400x)
+#### File Upload Methods
 
-#### Valid and Invalid Table Names
-Based on [PostgreSQL Documentation](https://www.postgresql.org/docs/13/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS), some naming conventions apply to objects, including tables, views, and columns. For most cases, we'll handle any necessary quoting automatically and transparently and generate unique names when required by adding a number as the last character.
+There are three main blocks that specifies three ways to upload the file/data for import. Each block has some important description with it. Each active way is highlighted when in use.
 
-- You can use any character for the valid characters except for a literal `null` character in a name.
-- For starting characters, anything goes (except a literal null) for our use case with the same caveats w.r.t. exotic characters above.
+##### Local File
 
-### User Selects Add Table Method
-After confirming the table name and schema, the user must select one of the methods available for adding a table. 
+The local file on the disk can be uploaded by dragging to the block or clicking on the block. Once the user drag the file to the block or select the file from their local machine, they can see the processing view.
 
-![](/assets/design/specs/table-import/chcPbaV.png =320x)
+##### URI
 
-Implementation Notes:
-- Copy will be revised later with the purpose of supporting best practices while guiding the user during the table creation process. We want to be more explicit in regards to goals for every step. 
+User can paste a URI in this block. If the URI is valid, the user would be able to click on the continue button to move to the processing import view.
 
-#### Implementation Note:
-- Empty table flow will be designed in a future iteration.
+##### Paste
 
-### User Selects Data Import Method
-Once the new table is created, The system will prompt the user to start the import data process. There are various options to import data, whether from a local file, clipboard, or remote location.
-By default, Upload File is the selected option.
+User can paste data directly here in this block. If the data pasted here is valid, the user would be able to click on the continue button to move to the processing import view.
 
-![](/assets/design/specs/table-import/zU9Csfe.png)
 
+#### Import Options
 
-### User Uploads a File
-The user then proceeds to upload a CSV file, and once it's completed, the system will render the file along with an option to remove it and upload another file.
+There are a few flags that user will find helpful for the import process. And they are present on the upload view itself. 
 
-![](/assets/design/specs/table-import/1FZ5Efd.png)
+##### Data Type Inference
 
```

Comment:
Okay I agree with this. This would put more cognitive load seeing our user persona. And we do not need this on the first screen since user is handling import here so we should let them focus on that rather than worrying about future errors caused by data inference.